chiselFrontend/src/main/scala/chisel3/core/Mem.scala

```diff
@@ -30,16 +30,30 @@ object Mem {
   }
 }
 
-sealed abstract class MemBase[T <: Data](t: T, val length: Int) extends HasId with VecLike[T] {
+/* The following definition of ChiselVecLike is to work around the issue that you cannot
```

Comment:
So basically move the `apply` macro into VecLike (so the transform happens with all VecLike subtypes), and define an abstract `do_apply` in VecLike (which is what the macro renames things into). `do_apply` is just a normal function, and can be overridden normally.